### ISSUE TITLE
Remote async with deferred launch policy never executes

### ISSUE THREAD

**Issue by K-ballo** [STEllAR-GROUP/hpx#1523]
Remote `async` does not behave correctly when given a `launch::deferred` policy. The action is not executed by `async` itself nor by waiting on the returned future. The following snippet hangs forever:

    void null_thread() {}
    HPX_PLAIN_ACTION(null_thread, null_action);

    int main() {
        hpx::async<null_action>(hpx::launch::deferred, hpx::find_here()).get();
        return 0;
    }
